### PR TITLE
feat: implement the drawer component of the new navigation

### PR DESCRIPTION
**Pull request by mkarajohn**
Resolves [DSTRBTN-540]

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request introduces a new Drawer component in the Navigation section of the UI. The Drawer component is a sidebar navigation menu that can be expanded or collapsed. It includes a client selector for switching between different organizations and a navigation header. 
> 
> ## What changed
> Three new files were created: `Drawer.styles.ts`, `Drawer.tsx`, and `index.ts` in the `src/ui/Navigation/components/Drawer/` directory. 
> 
> In `Drawer.styles.ts`, various styled components were created for the Drawer component, including the main container, organization switcher wrapper, navigation header, extras container, and navigation elements container.
> 
> In `Drawer.tsx`, the Drawer component was defined with props for expanded state, menu items, organization switching, selected organization, navigation header, extras, and desktop view. The component includes a client selector for switching organizations and a navigation header.
> 
> In `index.ts`, the Drawer component was exported.
> 
> ## How to test
> To test the changes, follow these steps:
> 
> 1. Pull the changes from this PR to your local environment.
> 2. Navigate to the `src/ui/Navigation/components/Drawer/` directory.
> 3. Open the `Drawer.tsx` file and check the Drawer component.
> 4. You can import the Drawer component in another file and use it with different props to see how it behaves.
> 5. Check the styled components in `Drawer.styles.ts` and see how they are used in the Drawer component.
> 
> ## Why make this change
> This change was made to provide a sidebar navigation menu for the application. The Drawer component improves the user experience by allowing users to easily navigate between different sections of the application and switch between different organizations. The expandable and collapsible feature of the Drawer makes it adaptable to different screen sizes, enhancing its usability.
</details>

[DSTRBTN-540]: https://orfium.atlassian.net/browse/DSTRBTN-540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #74** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/74" target="_blank">Graphite</a> 
    * **PR #75** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/75" target="_blank">Graphite</a> 
      * **PR #76** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/76" target="_blank">Graphite</a> 
        * **PR #77** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/77" target="_blank">Graphite</a> 
          * **PR #78** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/78" target="_blank">Graphite</a> 
            * **PR #79** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/79" target="_blank">Graphite</a> 
              * **PR #80** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/80" target="_blank">Graphite</a> 
                * **PR #81** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/81" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/81?utm_source=stack-comment).

File: src/ui/Navigation/components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  return (
+    <DrawerContainer
+      expanded={props.expanded}
+      isDesktop={isDesktop}
+      data-testid={'local-navigation'}
+    >
+      {hideOrgSwitcher ? null : (
+        <OrgSwitcherWrapper>
+          <ClientSelector
+            tagText={'Yolo'}
```
Comment (panvourtsis) [inline src/ui/Navigation/components/Drawer/Drawer.tsx:52]: is this a leftover?

Comment (mkarajohn) [reply src/ui/Navigation/components/Drawer/Drawer.tsx:52]: Yep, it's not there anymore 👀

**Review by panvourtsis — APPROVED** [review]

**Comment by mkarajohn** [comment]
### Merge activity

* **Oct 24, 10:43 AM**: @mkarajohn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/81).

**Pull request merged**